rllib/algorithms/ppo/ppo_rl_module.py

```diff
@@ -20,6 +20,9 @@ class PPORLModule(RLModule, abc.ABC):
     def setup(self):
         # __sphinx_doc_begin__
         catalog = self.config.get_catalog()
+        # If we have a stateful model states for the critic need to be collected
```

Comment:
Shouldn't we also use `is_stateful()` here? What if the user doesn't use the built-in use_lstm option, but comes with their own stateful model?

Reply:
@sven1977 this was my first intend, however at this point in time `is_stateful()` cannot be called, yet b/c the encoder is not yet defined.

Reply:
I agree that this is not a nice solution, but at this point in the code we need to know, if the module is stateful or not, but the `is_stateful()` depends on the encoder which is defined depending on `inference-only` being `True/False`.